community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/FusionSelector20.java

```diff
@@ -0,0 +1,125 @@
+import org.neo4j.internal.kernel.api.IndexQuery;
+import org.neo4j.storageengine.api.schema.IndexReader;
+import org.neo4j.values.storable.Value;
+import org.neo4j.values.storable.ValueGroup;
+import org.neo4j.values.storable.Values;
+
+import static org.neo4j.kernel.impl.index.schema.fusion.FusionIndexBase.LUCENE;
+import static org.neo4j.kernel.impl.index.schema.fusion.FusionIndexBase.NUMBER;
+import static org.neo4j.kernel.impl.index.schema.fusion.FusionIndexBase.SPATIAL;
+import static org.neo4j.kernel.impl.index.schema.fusion.FusionIndexBase.STRING;
+import static org.neo4j.kernel.impl.index.schema.fusion.FusionIndexBase.TEMPORAL;
+
+/**
+ * Selector for "lucene+native-2.x".
+ * Separates strings, numbers, temporal and spatial into native index.
+ */
+public class FusionSelector20 implements FusionIndexProvider.Selector
+{
+    @Override
+    public void validateSatisfied( Object[] instances )
+    {
+        FusionIndexBase.validateSelectorInstances( instances, STRING, NUMBER, SPATIAL, TEMPORAL, LUCENE );
+    }
+
+    @Override
+    public int selectSlot( Value... values )
+    {
+        if ( values.length > 1 )
+        {
+            // Multiple values must be handled by lucene
+            return LUCENE;
+        }
+
+        Value singleValue = values[0];
+        if ( singleValue.valueGroup() == ValueGroup.TEXT )
+        {
+            return STRING;
+        }
+
+        if ( singleValue.valueGroup() == ValueGroup.NUMBER )
+        {
+            return NUMBER;
+        }
+
+        if ( Values.isGeometryValue( singleValue ) )
+        {
+            return SPATIAL;
+        }
+
+        if ( Values.isTemporalValue( singleValue ) )
+        {
+            return TEMPORAL;
+        }
+
+        return LUCENE;
+    }
+
+    @Override
+    public IndexReader select( IndexReader[] instances, IndexQuery... predicates )
+    {
+        if ( predicates.length > 1 )
+        {
+            return instances[LUCENE];
+        }
+        IndexQuery predicate = predicates[0];
+
+        if ( predicate instanceof IndexQuery.ExactPredicate )
+        {
+            IndexQuery.ExactPredicate exactPredicate = (IndexQuery.ExactPredicate) predicate;
+            return select( instances, exactPredicate.value() );
+        }
+
+        if ( predicate instanceof IndexQuery.StringPredicate )
+        {
+            return instances[STRING];
+        }
+
+        if ( predicate instanceof IndexQuery.RangePredicate )
+        {
+            switch ( predicate.valueGroup() )
+            {
+            case NUMBER:
+                return instances[NUMBER];
+            case GEOMETRY:
+                return instances[SPATIAL];
+            case TEXT:
+                return instances[STRING];
+            case DATE:
+            case LOCAL_DATE_TIME:
+            case ZONED_DATE_TIME:
+            case LOCAL_TIME:
+            case ZONED_TIME:
+            case DURATION:
+                return instances[TEMPORAL];
+            default: // fall through
```

Comment:
This falling-through is confusing and error-prone; please throw UOE exception right here.

Reply:
Fixed in mentioned PR https://github.com/neo4j/neo4j/pull/11363